lldb/packages/Python/lldbsuite/test/lldbplatformutil.py

```diff
@@ -333,3 +335,41 @@ def expectedCompiler(compilers):
             return True
 
     return False
+
+
+# This is a helper function to determine if a specific version of Xcode's linker
+# contains a TLS bug. We want to skip TLS tests if they contain this bug, but
+# adding a linker/linker_version conditions to a decorator is challenging due to
+# the number of ways linkers can enter the build process.
+def darwinLinkerHasTLSBug():
```

Comment:
should there be two functions here? A generic function that can check for arbitrary version ranges, and then a specific `darwinLinkerHasTLSBug` which is implemented as:

```py
def darwinLinkerHasTLSBug():
    return 1000 <= checkDarwinLinkerVersion() <= 1109
```